sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1548,6 +1836,82 @@ submitted in its entirety for execution via
   ...
 ----
 
+=== Dynamic Parameter Update
+
+Example showing a graph with a single kernel node that is created using a kernel
+bundle with `handler::set_args()` and having its node arguments updated.
+
+[source,c++]
+----
+...
+
+using namespace sycl;
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+queue myQueue;
+auto myContext = myQueue.get_context();
+auto myDevice = myQueue.get_device();
+
+// USM allocations for kernel input/output
+const size_t n = 1024;
+int *ptrX = malloc_shared<int>(n, myQueue);
+int *ptrY = malloc_device<int>(n, myQueue);
+
+int *ptrZ = malloc_shared<int>(n, myQueue);
+int *ptrQ = malloc_device<int>(n, myQueue);
+
+// Kernel loaded from kernel bundle
+const std::vector<kernel_id> builtinKernelIds =
+      myDevice.get_info<info::device::built_in_kernel_ids>();
+kernel_bundle<bundle_state::executable> myBundle =
+      get_kernel_bundle(myContext, { myDevice }, builtinKernelIds);
+kernel builtinKernel = myBundle.get_kernel(builtinKernelIds[0]);
+
+// Graph containing a two kernels node
+sycl_ext::command_graph myGraph(myContext, myDevice);
+
+// Create graph dynamic parameters
+dynamic_parameter<int*> dynParamInput(myGraph);
+dynamic_parameter<int> dynParamScalar(myGraph);
```

Comment:
I think the examples should be easily findable from the spec.  So, if they are are not part of the spec, we should include a link to the other document.  For example, we could have two documents in the "sycl/doc/extensions/experimental" directory.

However, I don't see a problem with keeping the examples in the same document.  Yes, your spec is long, but it's not because of the examples.  By my rough estimate, the examples are currently about 10% of the document length, which isn't so bad.  Expanding the example "Dynamic Parameter Update" to illustrate how to update an accessor would add only a few lines.

Reply:
Apologies I don't think I explained it well since this work is still in the early stages. I agree the intention would be to link to the other document(s) from the spec where we think more detailed usage information or example code is useful, but is not included inline to avoid cluttering the spec.

>By my rough estimate, the examples are currently about 10% of the document length, which isn't so bad.

I would argue this is because we have conciously not added any more large examples to the spec because it would clutter it too much. However, there is a feeling that the extension could benefit from more example code and detailed usage guidance but that adding it to the spec would be deterimental to readability. The extension is large and also has many potential interactions with the main SYCL spec and there is definitely room for improving both of these aspects over what we already have.

My personal feeling would be that the spec could probably benefit from more small snippets, for example in this case perhaps a small snippet showing placeholder accessor usage, with more detailed or full examples such as the few we include at the end of the spec moved to separate linked document(s).

So it is not so much about adding a few lines to the example in this PR now, as much as I would like to focus on ironing out the actual API and reworking or improving the examples can be done as part of that ongoing work. Does that make sense?

Reply:
> My personal feeling would be that the spec could probably benefit from more small snippets, for example in this case perhaps a small snippet showing placeholder accessor usage

Yes, I agree with this also.

> So it is not so much about adding a few lines to the example in this PR now, as much as I would like to focus on ironing out the actual API and reworking or improving the examples can be done as part of that ongoing work.

I understand why you want to do it this way.  The examples help me to understand the APIs, though.  I don't mind if you add the examples later, but I may end up making new comments once the examples are there.

Reply:
That is a fair point that I had not considered, I'll take another look at the examples in this PR.

Reply:
I've add a separate smaller example that specifically shows usage with buffers/accessors.